contracts/upgradeable_contracts/modules/interest/AAVEInterestERC20.sol

```diff
@@ -0,0 +1,240 @@
+pragma solidity 0.7.5;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "../../../interfaces/IAToken.sol";
+import "../../../interfaces/IOwnable.sol";
+import "../../../interfaces/IInterestReceiver.sol";
+import "../../../interfaces/IInterestImplementation.sol";
+import "../../../interfaces/ILendingPool.sol";
+import "../MediatorOwnableModule.sol";
+
+/**
+ * @title AAVEInterestERC20
+ * @dev This contract contains token-specific logic for investing ERC20 tokens into AAVE protocol.
+ */
+contract AAVEInterestERC20 is IInterestImplementation, MediatorOwnableModule {
+    using SafeMath for uint256;
+
+    event PaidInterest(address indexed token, address to, uint256 value);
+
+    struct InterestParams {
+        IAToken aToken;
+        uint96 dust;
+        uint256 investedAmount;
+        address interestReceiver;
+        uint256 minInterestPaid;
+    }
+
+    mapping(address => InterestParams) public interestParams;
+
+    constructor(address _omnibridge, address _owner) MediatorOwnableModule(_omnibridge, _owner) {}
+
+    /**
+     * @dev Tells the module interface version that this contract supports.
+     * @return major value of the version
+     * @return minor value of the version
+     * @return patch value of the version
+     */
+    function getModuleInterfacesVersion()
+        external
+        pure
+        returns (
+            uint64 major,
+            uint64 minor,
+            uint64 patch
+        )
+    {
+        return (1, 0, 0);
+    }
+
+    /**
+     * @dev Tells the address of the LendingPool contract in the Ethereum Mainnet.
+     */
+    function lendingPool() public pure virtual returns (ILendingPool) {
+        return ILendingPool(0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9);
+    }
+
+    /**
+     * @dev Enables support for interest earning through specific aToken.
+     * @param _aToken address of the aToken contract. Underlying token address is derived from this contract.
+     * @param _dust small amount of underlying tokens that cannot be paid as an interest. Accounts for possible truncation errors.
+     * @param _interestReceiver address of the interest receiver for underlying token and associated COMP tokens.
+     * @param _minInterestPaid min amount of underlying tokens to be paid as an interest.
+     */
+    function enableInterestToken(
+        IAToken _aToken,
+        uint96 _dust,
+        address _interestReceiver,
+        uint256 _minInterestPaid
+    ) external onlyOwner {
+        address token = _aToken.UNDERLYING_ASSET_ADDRESS();
+        // disallow reinitialization of tokens that were already initialized and invested
+        require(interestParams[token].investedAmount == 0);
+
+        interestParams[token] = InterestParams(_aToken, _dust, 0, _interestReceiver, _minInterestPaid);
```

Comment:
Should we check `dust` for compliance with some limits? For example, can it be larger than one token (e.g. `1'000'000` for USDC)? What is our recommendation to have the value for `dust`?

Reply:
I don't think there is an easy way to define such limits for different tokens with different value and decimals. But in order to avoid possible misconfiguration, I have added a setter for dust parameter, so that we will be able change it if needed.

I think in most of the cases, a value of 1 token would be reasonable for this parameter (10**6 for USDC, 10**18 for DAI).